src/main/java/com/whalespottingjava/controllers/AdminController.java

```diff
@@ -24,8 +27,32 @@ public AdminController (SightingService sightingService) {
     @GetMapping("/admin")
     @ResponseStatus(HttpStatus.OK)
     public String getAdminPage(Model model) {
+        AdminApprovalRequests form = new AdminApprovalRequests();
+        model.addAttribute("form", form);
         model.addAttribute("sightings", sightingService.getAllPendingSightings());
+        return "admin";
+    }
+
+    @PostMapping("/admin")
+    @ResponseStatus(HttpStatus.CREATED)
+    public String postAdminApprovalRequest(@ModelAttribute AdminApprovalRequests form, Model model) {
+        int numbOfRequests = form.getAdminApprovalRequests().size();
+        for (AdminApprovalRequest adminApprovalRequest:form.getAdminApprovalRequests()) {
 
+            if (adminApprovalRequest.getId() == null || adminApprovalRequest.getApproved() == null) {
+                continue;
+            }
+            else if (adminApprovalRequest.getApproved().equals("true") && adminApprovalRequest.getId() != null) {
+                long sightingId = adminApprovalRequest.getId();
+                sightingService.approveSightingById(sightingId);
+            }
+            else if (adminApprovalRequest.getApproved().equals("false") && adminApprovalRequest.getId() != null) {
+                long sightingId = adminApprovalRequest.getId();
+                sightingService.deleteRejectedPendingSighting(sightingId);
+            }
+        }
```

Comment:
I feel like this part could be a bit shorter, there is too much check on the statements (e.g. you're checking if adminApprovalRequest is not null twice)

st like:

`if adminApproval.getId() != null { if (approved.rquals(true))}`

Reply:
also, continue does nothing in here because its not a loop. I know I put it there but I had a googling on that and it doesn't behave the way you want to achieve. I guess better to keep that if statement in blank or throw an exception

Reply:
Yes, I agree